### PR TITLE
Add delete into memory.md

### LINKED ISSUE #1736: Update documentation on memory
While answering a question from @xDiesel76 on our matrix channel, I realised that the documentation on opsdroid memory is missing some things like the [`delete` method](https://github.com/opsdroid/opsdroid/blob/d92a129c557d209172fb83507a3e1a414fb8e28a/opsdroid/memory.py#L53).

Just to make sure that this documentation stays up to date with the code, I would use the sphinx autodoc to generate the documentation from the docstrings. For example:

```
```eval_rst
.. autoclass:: opsdroid.memory.Memory
    :members:
```

Feel free to ask for help if you need a hand with sphinx autodoc!

Jacob also mentioned on the chat that you could grab the database connection with `self.opsdroid.get_database("sqlite").client`

### PR DESCRIPTION
**Pull request by xDiesel76**
# Description

Added delete 

Fixes #1736

## Status

**READY**

## Type of change

- Documentation (fix or adds documentation)

# Checklist:

- [Y] I have performed a self-review of my own code
- [Y] I have made corresponding changes to the documentation (if applicable)
- [Y] I have added tests that prove my fix is effective or that my feature works
- [Y] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/skills/memory.md [hunk 1]
```diff
@@ -14,6 +14,10 @@ Returns an object from the memory for the key provided.
 
 Stores the object provided for a specific key.
 
+### `delete(key)`
+
+Deletes the object provided the specific key.
+
 ### Example
```
Comment (FabioRosado) [inline docs/skills/memory.md:21]: Let's move this example up, we can replace the `Persisting data` header with example one

**Review by FabioRosado — CHANGES_REQUESTED** [review]
Thank you for working on this, this is great! I've commented on the [open issue](https://github.com/opsdroid/opsdroid/issues/1736#issuecomment-772749317)

But it could be good to replace this and let sphinx generate the methods instead.

File: docs/skills/memory.md [hunk 2]
```diff
@@ -32,8 +36,19 @@ class RememberSkill(Skill):
...
-In the above example we have defined two skill functions. The first takes whatever the user says after the word "remember" and stores it in the database.
+In the above example we have defined three skill functions. The first takes whatever the user says after the word "remember" and stores it in the database.
 
 The second retrieves and prints out that text when the user says "remind me".
+
+The third deletes what is remembered in the database when the user says "forget it".
+
+```eval_rst
+.. autoclass:: opsdroid.memory.Memory
+    :members:
```
Comment (FabioRosado) [inline docs/skills/memory.md:54]:
```suggestion
    :members:
    ```
```

File: docs/skills/memory.md [hunk 3]
```diff
@@ -32,8 +36,19 @@ class RememberSkill(Skill):
...
+        await message.respond("Ok I'll forget it")
 ```
 
-In the above example we have defined two skill functions. The first takes whatever the user says after the word "remember" and stores it in the database.
+In the above example we have defined three skill functions. The first takes whatever the user says after the word "remember" and stores it in the database.
 
 The second retrieves and prints out that text when the user says "remind me".
+
+The third deletes what is remembered in the database when the user says "forget it".
+
```
Comment (FabioRosado) [inline docs/skills/memory.md:51]:
```suggestion
## Reference

```

**Review by FabioRosado — CHANGES_REQUESTED** [review]
Thank you for the update, I've added some suggestions, mostly to Add a second heading with `Reference` so we can display the memory autodocs. I couldn't see the closure of the backticks, not sure if this is some issue on my end or if they are missing

File: docs/skills/memory.md [hunk 4]
```diff
@@ -32,8 +36,21 @@ class RememberSkill(Skill):
...
 
 The second retrieves and prints out that text when the user says "remind me".
+
+The third deletes what is remembered in the database when the user says "forget it".
+## Reference
+
+```eval_rst
+.. autoclass:: opsdroid.memory.Memory
+    :members:
+    ```
```
Comment (FabioRosado) [inline docs/skills/memory.md:56]: Sorry to be a pain, you will need to move these backticks to the beginning of the line yourself, I couldn't do it when making the suggestion

Comment (xDiesel76) [reply docs/skills/memory.md:56]:
```eval_rst
.. autoclass:: opsdroid.memory.Memory
    :members:
```
Like this?

Comment (FabioRosado) [reply docs/skills/memory.md]: Yeah haha

**Review by FabioRosado — APPROVED** [review]
Thank you for the update on this, I've tested things locally, and they all work as expected! We now have a nice reference to the memory docs as well, which is great!

I'm merging this PR now 😄 👍

**Pull request merged**